both/collections/csvdata.collection.ts

```diff
@@ -1,159 +1,317 @@
+                    "Assigned_category": "not assigned",
+                    "is_processed": 0,
+                    "invoice_no": "not_assigned",
+                    "invoice_description": "invoice_description",
+                    "Assigned_user_id": "not_assigned",
+                    "Assigned_username": "not_assigned"
+                });
+                if (exists_graph) {
+                    var obj = {};
+                    if (item["Cr/Dr"] == "CR") {
+                        obj[month[month_value] + '_CR'] = amount;
+                    } else {
+                        obj[month[month_value] + '_DR'] = amount;
+                    }
+
+                    Graphdata.update({
+                        "year": year
+                    }, {
+                        $inc: obj
+                    });
+
+                } else {
+                    var obj = {};
+                    if (item["Cr/Dr"] == "CR") {
+                        obj[month[month_value] + '_CR'] = amount;
+                    } else {
+                        obj[month[month_value] + '_DR'] = amount;
+                    }
+                    Graphdata.insert({
+                        "year": year,
+                        "January_CR": 0,
+                        "February_CR": 0,
+                        "March_CR": 0,
+                        "April_CR": 0,
+                        "May_CR": 0,
+                        "June_CR": 0,
+                        "July_CR": 0,
+                        "August_CR": 0,
+                        "September_CR": 0,
+                        "October_CR": 0,
+                        "November_CR": 0,
+                        "December_CR": 0,
+                        "January_DR": 0,
+                        "February_DR": 0,
+                        "March_DR": 0,
+                        "April_DR": 0,
+                        "May_DR": 0,
+                        "June_DR": 0,
+                        "July_DR": 0,
+                        "August_DR": 0,
+                        "September_DR": 0,
+                        "October_DR": 0,
+                        "November_DR": 0,
+                        "December_DR": 0
+                    });
+
+                    Graphdata.update({
+                        "year": year
+                    }, {
+                        $inc: obj
+                    });
+                }
+
+
+            }
+        }
+        return true;
+    }, // Meteor method addcategory will assign category to our document which we choose in csvjson component
+
+    'addCategory' (id, category) {
+        check(id, String);
+        check(category, String);
+        Csvdata.update({
+            "_id": id
+        }, {
+            $set: {
+                "Assigned_category": category,
+                "is_processed": 1
+            }
+        });
+    },
+    'changeCategory' (id, category) {
+        check(id, String);
```

Comment:
change category is use if we want to change a category that is assigned to any transaction note. now only admin can change this category